src/Date.php

```diff
@@ -75,7 +75,16 @@ public static function make($time = null, $timezone = null)
      */
     public static function parse($time = null, $timezone = null)
     {
-        $time = static::translateTimeString($time);
+        if ($time instanceof Carbon) {
+            return new static(
+                $time->toDateTimeString(),
+                $timezone ?: $time->getTimezone()
+            );
+        }
+
+        if (! is_int($time)) {
```

Comment:
If `$time` is an integer, it's a Unix timestap. By not sending it through `translateTimeString()`, it remains an integer and is then correctly handled in the constructor.